llvm/test/SYCLLowerIR/esimd_global.ll

```diff
@@ -0,0 +1,166 @@
+  %group.id.y.i.i.i.i.i = call i32 @llvm.genx.group.id.y() #5
+  %group.id.y.cast.ty.i.i.i.i.i = zext i32 %group.id.y.i.i.i.i.i to i64
+  %mul.i.i.i.i.i = mul nuw i64 %wgsize.y.cast.ty.i.i.i.i.i, %group.id.y.cast.ty.i.i.i.i.i
+  %add.i.i.i.i.i = add i64 %mul.i.i.i.i.i, %local_id.y.cast.ty.i.i.i.i.i
+  %local_id.x.i.i.i.i.i = extractelement <3 x i32> %call.esimd.i.i.i.i.i, i32 0
+  %local_id.x.cast.ty.i.i.i.i.i = zext i32 %local_id.x.i.i.i.i.i to i64
+  %wgsize.x.i.i.i.i.i = extractelement <3 x i32> %call.esimd1.i.i.i.i.i, i32 0
+  %wgsize.x.cast.ty.i.i.i.i.i = zext i32 %wgsize.x.i.i.i.i.i to i64
+  %group.id.x.i.i.i.i.i = call i32 @llvm.genx.group.id.x() #5
+  %group.id.x.cast.ty.i.i.i.i.i = zext i32 %group.id.x.i.i.i.i.i to i64
+  %mul.i4.i.i.i.i = mul nuw i64 %group.id.x.cast.ty.i.i.i.i.i, %wgsize.x.cast.ty.i.i.i.i.i
+  %add.i5.i.i.i.i = add i64 %mul.i4.i.i.i.i, %local_id.x.cast.ty.i.i.i.i.i
+  %0 = bitcast %"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd"* %agg.tmp.i to i8*
+  call void @llvm.lifetime.start.p0i8(i64 64, i8* nonnull %0)
+  %1 = bitcast %"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd"* %vc.i to i8*
+  call void @llvm.lifetime.start.p0i8(i64 64, i8* nonnull %1) #5
+  %conv.i = trunc i64 %add.i5.i.i.i.i to i32
+  %2 = addrspacecast %"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd"* %vc.i to %"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd" addrspace(4)*
+  %splat.splatinsert.i.i = insertelement <16 x i32> undef, i32 %conv.i, i32 0
+  %splat.splat.i.i = shufflevector <16 x i32> %splat.splatinsert.i.i, <16 x i32> undef, <16 x i32> zeroinitializer
+  %M_data.i13.i = getelementptr inbounds %"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd", %"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd" addrspace(4)* %2, i64 0, i32 0
+  store <16 x i32> %splat.splat.i.i, <16 x i32> addrspace(4)* %M_data.i13.i, align 64, !tbaa !13
+  %conv3.i = trunc i64 %add.i.i.i.i.i to i32
+  %splat.splatinsert.i20.i = insertelement <8 x i32> undef, i32 %conv3.i, i32 0
+  %splat.splat.i21.i = shufflevector <8 x i32> %splat.splatinsert.i20.i, <8 x i32> undef, <8 x i32> zeroinitializer
+  %call.esimd.i.i.i.i.i2 = call <16 x i32> @llvm.genx.vload.v16i32.p4v16i32(<16 x i32> addrspace(4)* %M_data.i13.i) #5
+  %call4.esimd.i.i.i.i = call <16 x i32> @llvm.genx.wrregioni.v16i32.v8i32.i16.v8i1(<16 x i32> %call.esimd.i.i.i.i.i2, <8 x i32> %splat.splat.i21.i, i32 0, i32 8, i32 1, i16 0, i32 0, <8 x i1> <i1 true, i1 true, i1 true, i1 true, i1 true, i1 true, i1 true, i1 true>) #5
+  call void @llvm.genx.vstore.v16i32.p4v16i32(<16 x i32> %call4.esimd.i.i.i.i, <16 x i32> addrspace(4)* %M_data.i13.i) #5
+  %cmp.i = icmp eq i64 %add.i.i.i.i.i, 0
+  %..i = select i1 %cmp.i, i64 %add.i5.i.i.i.i, i64 %add.i.i.i.i.i
+  %conv9.i = trunc i64 %..i to i32
+; CHECK: store <16 x i32> {{.+}} bitcast (<16 x i32>* [[NEWGLOBAL]] to {{.+}}
```

Comment:
It seems like some interesting things are missing from the check here. For instance, the original store contains an addressspace cast. What happens to that?

Reply:
updated test to check for addrspacecasts